### PR TITLE
internal/v5: resource upload requires pull as well as push

### PR DESCRIPTION
**Pull request by rogpeppe**
Also make it harder to create a config file with a key or certificate
that is non-empty but doesn't contain a valid key or cert,
and actually pass the docker registry address through to the server.

### CODE REVIEW

File: internal/dockerauth/api.go [hunk 1]
```diff
@@ -115,6 +115,7 @@ func (h *handler) Token(p httprequest.Params, req *tokenRequest) (*tokenResponse
 	}
 	ms := credentials(p.Request)
 	filteredRAs := make([]resourceAccessRights, 0, len(ras))
+	logger.Infof("requested access rights: %#v", ras)
```
Comment (mhilton) [inline internal/dockerauth/api.go]: This looks like debugging.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore/2211/

File: internal/dockerauth/api.go [hunk 2]
```diff
@@ -129,7 +130,7 @@ func (h *handler) Token(p httprequest.Params, req *tokenRequest) (*tokenResponse
 			if err == nil {
 				filteredActions = append(filteredActions, a)
 			} else {
-				logger.Infof("check failed: %v", err)
+				logger.Infof("docker token check failed for operation %q: %v", a, err)
```
Comment (mhilton) [inline internal/dockerauth/api.go]: I'm not certain this should be logged at info level either.

**Review by mhilton — APPROVED** [review]
LGTM with a couple of thoughts.

**Review by fabricematrat — APPROVED** [review]

File: internal/dockerauth/api.go (hunk 2 above)
Comment (rogpeppe) [reply internal/dockerauth/api.go]: changed to Debug.

File: internal/dockerauth/api.go (hunk 1 above)
Comment (rogpeppe) [reply internal/dockerauth/api.go]: removed

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore/2212/

**Pull request merged**